### PR TITLE
Bug: keep track of latest selected offer on PDP

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Refresh selected offer after editing price intent/offer (start date or re-confirm).

## Justify why they are needed

We have a bug now where you don't see the updated start date after you edit. This is because we store the selected offer in local state and don't update it after editing. This PR fixes that.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2010** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2010" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2010?utm_source=stack-comment).

File: apps/store/src/components/ProductPage/PurchaseForm/OfferPresenter.tsx [hunk 1]
```diff
@@ -37,9 +39,10 @@ type Props = {
 
 export const OfferPresenter = (props: Props) => {
   const { priceIntent, shopSession, scrollPastRef, onAddedToCart, onClickEdit } = props
+  const { selectedOffer } = props
```
Comment (alebedev) [inline apps/store/src/components/ProductPage/PurchaseForm/OfferPresenter.tsx:42]: Maybe we should read it from atom instead of passing via props?

**Review by alebedev — APPROVED** [review]
Looks good

File: apps/store/src/components/ProductPage/PurchaseForm/OfferPresenter.tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/components/ProductPage/PurchaseForm/OfferPresenter.tsx:42]: yeah but then we need to deal with a nullable value

the offer presenter should never be rendered with without a selected offer so I chose to pass it in instead

makes sense?

File: apps/store/src/components/ProductPage/PurchaseForm/OfferPresenter.tsx [hunk 2]
```diff
@@ -37,9 +39,10 @@ type Props = {
...
+  const [, setSelectedOffer] = useSelectedOffer()
```
Comment (guilhermespopolin) [inline apps/store/src/components/ProductPage/PurchaseForm/OfferPresenter.tsx:43]: Woudn't be the case of using Jotai's `useAtomValue` here?
I understand we're using Jotai via a custom hook..maybe we should export a read only option from `useSelectedOffer` as well? 🤔

**Review by guilhermespopolin — APPROVED** [review]

**Pull request merged**